src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,257 @@
+                        applyChangesOperation.ChangedSolution.GetAnalyzerConfigDocument, solution.GetAnalyzerConfigDocument,
+                        cancellationToken).ConfigureAwait(false);
+
+                    // Changed additional documents
+                    // We won't get any results until https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // is fixed.
+                    await GetTextDocumentEdits(
+                        textDocumentEdits, applyChangesOperation, solution, changedDocuments,
+                        applyChangesOperation.ChangedSolution.GetAdditionalDocument, solution.GetAdditionalDocument,
+                        cancellationToken).ConfigureAwait(false);
+                }
+
+                return textDocumentEdits.ToArray();
+
+                static async Task GetTextDocumentEdits<T>(
+                    ArrayBuilder<TextDocumentEdit> textDocumentEdits,
+                    ApplyChangesOperation applyChangesOperation,
+                    Solution solution,
+                    IEnumerable<DocumentId> changedDocuments,
+                    Func<DocumentId, T?> getNewDocumentFunc,
+                    Func<DocumentId, T?> getOldDocumentFunc,
+                    CancellationToken cancellationToken)
+                    where T : TextDocument
+                {
+                    foreach (var docId in changedDocuments)
+                    {
+                        var newDoc = getNewDocumentFunc(docId);
+                        var oldDoc = getOldDocumentFunc(docId);
+                        if (oldDoc == null || newDoc == null)
+                        {
+                            continue;
+                        }
+
+                        var oldText = await oldDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var newText = await newDoc.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                        var textChanges = newText.GetTextChanges(oldText).ToList();
+
+                        var edits = textChanges.Select(tc => ProtocolConversions.TextChangeToTextEdit(tc, oldText)).ToArray();
+                        var documentIdentifier = new VersionedTextDocumentIdentifier() { Uri = newDoc.GetURI() };
+                        textDocumentEdits.Add(new TextDocumentEdit() { TextDocument = documentIdentifier, Edits = edits.ToArray() });
+                    }
+                }
+            }
+
+            static LSP.Command SetCommand(VSCodeAction codeAction, CodeActionResolveData data) => new LSP.Command
+            {
+                CommandIdentifier = CodeActionsHandler.RunCodeActionCommandName,
+                Title = codeAction.Title,
+                Arguments = new object[] { data }
+            };
+        }
+
+        internal static CodeAction? GetCodeActionToResolve(string distinctTitle, IEnumerable<CodeAction> codeActions)
```

Comment:
please don't pass IEnumerables around.  It is impossible to tell if these ar elazy, and if iterating over them multiple times will hav a high perf cost.  use ImmutableArray.